src/Vm.sol

```diff
@@ -183,50 +256,97 @@ interface VmSafe {
     // ----
     // Given a string of JSON, return it as ABI-encoded
     function parseJson(string calldata json, string calldata key) external pure returns (bytes memory abiEncodedData);
+
     function parseJson(string calldata json) external pure returns (bytes memory abiEncodedData);
 
+    // The following parseJson cheatcodes will do type coercion, for the type that they indicate.
+    // For example, parseJsonUint will coerce all values to a uint256. That includes stringified numbers '12'
+    // and hex numbers '0xEF'.
+    // Type coercion works ONLY for discrete values or arrays. That means that the key must return a value or array, not
+    // a JSON object.
+
+    function parseJsonUint(string calldata, string calldata) external returns (uint256);
+
```

Comment:
And we should remove the blank lines between all these new cheats too

Reply:
This is the result of running `forge fmt`. It's more legible this way, no?

Reply:
Maybe you're on an old forge version or have some custom config? But for me and in CI `forge fmt` doesn't introduce line breaks. Just pushed dac8350 to match the original formatting for now to reduce the diff (give that a review to make sure I didn't accidentally remove anything, then we can merge). Open to organizing/cleaning up the `Vm` file in a separate issue/PR

Reply:
@mds1 uuuh, didn't think about that. Probs you are right. lgtm